tests/perf/test_accuracy.py

```diff
@@ -2,142 +2,255 @@
+            # Timm ImageNet model
+            mean_values = [(item * 255) for item in model_cfg["mean"]]
+            scale_values = [(item * 255) for item in model_cfg["std"]]
+            self.preprocess_fn = get_preprocess_fn(
+                change_channel_order=True,
+                input_size=model_cfg["input_size"][1:],
+                mean=mean_values,
+                std=scale_values,
+                hwc_to_chw=True,
+            )
+            self.postprocess_fn = get_postprocess_fn(activation=ActivationType.SOFTMAX)
+        else:
+            raise ValueError(f"Model {self.model_name} is not supported since it's not VOC or ImageNet model.")
+
+    def setup_explainer(self, model, explain_method):
+        explain_mode = ExplainMode.WHITEBOX if explain_method in WhiteBoxXAIMethods else ExplainMode.BLACKBOX
+
+        if self.model_name in TRANSFORMER_MODELS and explain_method == Method.RECIPROCAM:
+            explain_method = Method.VITRECIPROCAM
 
         self.explainer = Explainer(
             model=model,
             task=Task.CLASSIFICATION,
             preprocess_fn=self.preprocess_fn,
-            explain_mode=ExplainMode.WHITEBOX,
+            postprocess_fn=self.postprocess_fn,
+            explain_mode=explain_mode,
+            explain_method=explain_method,
+            embed_scaling=True,
         )
+        kwargs = {}
+        if explain_method in BlackBoxXAIMethods:
+            # TODO: Make Preset configurable as well
+            kwargs.update({"preset": Preset.SPEED})
+        return kwargs
+
+    @pytest.fixture(autouse=True)
+    def setup(self, fxt_output_root, fxt_data_root, fxt_dataset_parameters):
+        self.data_dir = fxt_data_root
+        # self.output_dir = fxt_output_root
+        self.output_dir = Path("/home/gzalessk/code/openvino_xai/tests/perf/validation_results/5000_subset_42_seed")
+        self.supported_num_classes = {1000: 1000}
+
+        self.setup_dataset(fxt_dataset_parameters)
+        self.dataset_name = "coco" if self.dataset_type == DatasetType.COCO else "voc"
+
+    @pytest.mark.parametrize("model_id", TEST_MODELS)
+    @pytest.mark.parametrize("explain_method", EXPLAIN_METHODS)
+    def test_explainer_images(self, model_id, explain_method):
+        self.model_name = model_id
+        self.data_metric_path = self.output_dir / self.model_name / explain_method.value
+        os.makedirs(self.data_metric_path, exist_ok=True)
+
+        model, model_cfg = self.setup_model(self.data_dir, self.model_name)
+        self.setup_process_fn(model_cfg)
+        black_box_kwargs = self.setup_explainer(model, explain_method)
 
         self.pointing_game = PointingGame()
         self.auc = InsertionDeletionAUC(model, self.preprocess_fn, self.postprocess_fn)
-        self.adcc = ADCC(model, self.preprocess_fn, self.postprocess_fn, self.explainer)
+        self.adcc = ADCC(model, self.preprocess_fn, self.postprocess_fn, self.explainer, **black_box_kwargs)
```

Comment:
why `black_box_kwargs` only for ADCC, what about `InsertionDeletion`?

Reply:
`black_box_kwargs` is needed to run explainer and get explanations, which is required only for `ADCC` metric